### ISSUE TITLE
Muon ID recipe for ICHEP

### ISSUE THREAD

**Issue by bianchini** [vhbb/cmssw#481]
See https://hypernews.cern.ch/HyperNews/CMS/get/higgs/1856.html

**Comment by bianchini** [comment]
And https://hypernews.cern.ch/HyperNews/CMS/get/higgs/1868.html

**Comment by GLP90** [comment]
Concerning the data trigger efficiency: there are already some preliminary numbers provided by the POG on 2016 data:
https://twiki.cern.ch/twiki/bin/view/CMS/MuonWorkInProgressAndPagResults#Results_on_7_6_fb
The current issue seems to be a drop in efficiency in large pT (>120) - large eta (>0.9). 

Another option would be to use the data efficiencies measured on 2015 data, which however stop at 120 GeV. I could provide the JSON in the afternoon.

**Comment by degrutto** [comment]
Hi Gael , what about at low pt? I guess the tracker problem (HIP) in 2016 would start to impact  muon effciency already  20 GeVish

**Comment by GLP90** [comment]
To take the tracker inefficiency into account, we should also include the following SF:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/MuonReferenceEffsRun2#Tracking_efficiency_provided_by

**Comment by GLP90** [comment]
Stephane noticed the IsoMu22_OR_IsoTkMu22 triggers are not available in the Ntuples for data.
How should we proceed, use the IsoMu20_OR_IsoTkMu20 triggers together with the IsoMu22_OR_IsoTkMu22 SF, and start use a pT > 24 GeV cut for the muons ?

**Comment by scooperstein** [comment]
About the muon ID and isolation, did we apply in V23 the 80X working points or the old 76X ones? If we used the new 80X working points then we can use the preliminary ID+ISO scale factors that Gael references above. If not, I guess we will have to discuss the best prescription to use for the moment.

**Comment by bianchini** [comment]
Ciao, we are still using old muon/electron ID's, as no action was taken in time for this issue. At this point any SF is just a placeholder for me. Could someone commit the new muon ID and the new SF?

**Comment by GLP90** [comment]
Ciao Lorenzo. I can take care about this. Could you point me to the part of the code where the ID's should be implemented ?

**Comment by GLP90** [comment]
Looking at this twiki from the muon POG:
https://twiki.cern.ch/twiki/bin/view/CMS/SWGuideMuonIdRun2#Short_Term_Instructions_for_ICHE
It seems the loose and tight ID definitions (used by Zll, Wln resp.) were not modified to handle the HIP. Does it mean our current implementation of the ID in heppy is correct ?

**Comment by scooperstein** [comment]
From what I see here in the Heppy code:

https://github.com/vhbb/cmssw/blob/vhbbHeppy80X/VHbbAnalysis/Heppy/test/vhbb.py#L322
https://github.com/vhbb/cmssw/blob/vhbbHeppy80X/VHbbAnalysis/Heppy/test/vhbb.py#L326

I think that we take directly from the 80X miniAOD the definition of the ID. If that is indeed the case I think we already handle it correctly. Can someone please verify?

**Comment by bianchini** [comment]
We are, but I doubt that the miniAOD scheduled long tome ago fulfill the https://twiki.cern.ch/twiki/bin/view/CMS/SWGuideMuonIdRun2#Short_Term_Medium_Muon_Definitio criteria...
I think one rather has to apply a few cuts by hand.

**Comment by GLP90** [comment]
Just for me to understand, why is it necessary ? Our selections are not using the Medium ID.

**Comment by scooperstein** [comment]
I think that Lorenzo is saying that the definition of the loose/tight ID might have anyway changed a bit between when the 80X campaign was submitted and now.

**Comment by GLP90** [comment]
Ok thanks. I will have a look on the implementation to check if it is up-to-date with the current definitions.

**Comment by GLP90** [comment]
Update: I have implemented the latest loose, medium and tight POG id from [1](the implementation are available in [2]).
I produced a set of 800 event with Vtype == 0. 
After comparing the *POG id (which are the one currently used for the tree preselection) and the new POG_ICHEP2016 ids, the conclusion are the following:

The current and ICHEP definitions for the loose and tight id are identical.
The current and ICHEP definitions for the medium id give different results, which is clear from [1].
This means the current V23 trees are using the correct, most-up-to-date ids (loose and tight). The only modification one could add for the next production is to update the medium id definition. I can make a PR for this, let me know.

[1]https://twiki.cern.ch/twiki/bin/view/CMS/SWGuideMuonIdRun2#Short_Term_Medium_Muon_Definitio
[2] https://github.com/GLP90/cmssw/commit/4f245a6b8f1e717bf4e5d1f81980948a29ec4a1f#diff-5acefa3545acbc95dd11bcce0f144441R57
